### PR TITLE
Fetching Events in CalendarView

### PR DESCRIPTION
**Pull request by Prkhar05**

### CODE REVIEW

**Comment by Prkhar05** [comment]
According  to Issue #59 ,Fetching Events from backend API and updated the event creation modal , also updated filters.

File: src/components/Calendar/CalenderComponent/EventModal.tsx [hunk 1]
```diff
@@ -1,29 +1,90 @@
...
+        }
+    
+        const year = dayjsObject.year();
+        const month = dayjsObject.month(); 
+        const day = dayjsObject.date();
+    
+        const dateObject = new Date(year, month, day, hour, minute);
+        return dateObject;
+    }
+    function generateUUID() {
```
Comment (mittal-ishaan) [inline src/components/Calendar/CalenderComponent/EventModal.tsx:65]: use `cuid` library.

File: src/components/Calendar/CalenderComponent/EventModal.tsx [hunk 2]
```diff
@@ -1,29 +1,90 @@
...
+    const [role,setRole] = useState(selectedEvent ? selectedEvent.job.role : "")
+    const [recruitmentType,setRecruitmentType] = useState(selectedEvent ? selectedEvent.job.season.type : "")
+    const [roundNumber,setRoundNumber] = useState(selectedEvent ? selectedEvent.roundNumber : 1)
+    const[IsOpen,setIsOpen] = useState<boolean>(false)
+
+    function displayColor(label:string){
+        return labelsClasses.get(label);
+      }
+
+    function convertToDateTime(timeString, dayjsObject) {
```
Comment (mittal-ishaan) [inline src/components/Calendar/CalenderComponent/EventModal.tsx:40]: there might be a JS libarary for it. please see if we can use that. the code is prone to errors if we do this ourselves

File: src/components/Calendar/context/ContextWrapper.tsx [hunk 3]
```diff
@@ -1,111 +1,166 @@
...
+
+function updateLabel(newLabel) {
+  setLabels((prevLabels) => prevLabels.map((lbl) => (lbl.label === newLabel.label ? newLabel : lbl)));
+}
 
-    function updateLabel(newLabel:updateLabelAction){
-      setLabels(labels.map((lbl) => lbl.label === newLabel.label ? newLabel : lbl))
+  async function addEvent(event:selectedDayEvent) {
+    try {
+      const response = await fetch('http://localhost:5000/api/v1/events', {
```
Comment (mittal-ishaan) [inline src/components/Calendar/context/ContextWrapper.tsx:91]: use the backend api from env and if it is not there add it there and use it

File: src/components/Calendar/context/ContextWrapper.tsx [hunk 4]
```diff
@@ -1,111 +1,166 @@
...
+      const data = await response.json();
+      dispatchCallEvents({ type: 'update', payload: data });
+    } catch (error) {
+      console.error('Failed to update event:', error);
     }
+  }
 
+  async function deleteEvent(eventId) {
+    try {
+      await fetch(`http://localhost:5000/api/v1/events/${eventId}`, {
```
Comment (mittal-ishaan) [inline src/components/Calendar/context/ContextWrapper.tsx:122]: same here

**Pull request merged**